datafusion/common/src/dfschema.rs

```diff
@@ -206,7 +206,7 @@ impl DFSchema {
                 (Some(qq), None) => {
                     // the original field may now be aliased with a name that matches the
                     // original qualified name
-                    let table_ref: TableReference = field.name().as_str().into();
+                    let table_ref = TableReference::parse_str(field.name().as_str());
```

Comment:
I am not sure if this is correct as it potentially splits on "." -- but I chose to keep the existing behavior